### PR TITLE
fix: set conda as the only python provider

### PR DESCRIPTION
**Pull request by lalawuu**
Signed-off-by: Qi Chen <aseaday@hotmail.com>

https://github.com/tensorchord/envd/issues/581
https://github.com/tensorchord/envd/issues/580

### CODE REVIEW

**Review by gaocegege — COMMENTED** [review]
Should we remove system python in the base image?

**Comment by aseaday** [comment]
> Should we remove system python in the base image?

I think it is ok.

File: base-images/python3.8-ubuntu20.04.Dockerfile [hunk 1]
```diff
@@ -25,7 +25,7 @@ RUN apt-get update && \
     bzip2 ca-certificates libglib2.0-0 libsm6 libxext6 libxrender1 mercurial \
     procps subversion wget \
     # envd dependencies
-    python3 curl openssh-client git tini sudo python3-pip zsh vim \
+    curl openssh-client git tini sudo python3-pip zsh vim \
```
Comment (gaocegege) [inline base-images/python3.8-ubuntu20.04.Dockerfile:28]: Also remove the python3-pip here please

Comment (aseaday) [reply base-images/python3.8-ubuntu20.04.Dockerfile]: I split it in two commit. sry

File: pkg/lang/ir/python.go [hunk 2]
```diff
@@ -24,6 +24,17 @@ import (
...
 
+func (g Graph) GetAppropriatePythonVersion() (string, error) {
+	version := *g.Language.Version
+	if version == "3" || version == "" {
+		return defaultVersion, nil
+	}
+	if strings.HasPrefix(version, "3.") {
+		return version, nil
+	}
+	return "", errors.Errorf("python version %s is not supported", version)
```
Comment (gaocegege) [inline pkg/lang/ir/python.go:35]: We do not support Python 2.x, does it work for you? @VoVAllen @kemingy

File: base-images/python3.8-ubuntu20.04.Dockerfile (hunk 1 above)
Comment (gaocegege) [reply base-images/python3.8-ubuntu20.04.Dockerfile]: LGTM!

Comment (aseaday) [reply base-images/python3.8-ubuntu20.04.Dockerfile]: Should I also rename the Dockerfile? maybe python-ubuntu20.04? But I think I need help to fix other hardcode such as image base in the code and CI process.

File: pkg/lang/ir/python.go (hunk 2 above)
Comment (aseaday) [reply pkg/lang/ir/python.go:35]: I forget to discuss with others. But I do think it is not a good idea to support python2. >_<!

**Comment by gaocegege** [comment]
Please fix the lint issues

File: pkg/lang/ir/python.go (hunk 2 above)
Comment (kemingy) [reply pkg/lang/ir/python.go:35]: LGTM

**Review by kemingy — APPROVED** [review]

File: pkg/lang/ir/python.go (hunk 2 above)
Comment (VoVAllen) [reply pkg/lang/ir/python.go:35]: LGTM. Python 2.x should be deprecated in all ways

**Pull request merged**